### ISSUE TITLE
[Bug]: UserLocation Componente isn't working

### ISSUE THREAD

**Issue by RayanCardoso** [rnmapbox/maps#3083]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### Platform

Android

### `@rnmapbox/maps` version

10.0.15

### Standalone component to reproduce

```javascript
import React, {useContext, useEffect, useRef, useState} from 'react';
import Mapbox from '@rnmapbox/maps';
import {
  Alert,
  Dimensions,
  FlatList,
  Image,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {TouchableOpacity} from 'react-native';
//import Geolocation from 'react-native-geolocation-service'; -> library I have used, placing here if have some incompatibility 

const accessToken = 'KEY_NAME';
Mapbox.setWellKnownTileServer('Mapbox');
Mapbox.setAccessToken(accessToken);

<View>
  <Mapbox.MapView
    preferredFramesPerSecond={120}>
    <Mapbox.Camera 
        zoomLevel={10} 
    />

    <Mapbox.UserLocation
      visible={true}
      animated={true}
      showsUserHeadingIndicator={true}
      androidRenderMode={'compass'}
    >
    </Mapbox.UserLocation>
  </Mapbox.MapView>
</View>
  


```

### Observed behavior and steps to reproduce

I attempted to build a Mapbox component for application but encountered an issue with an "invalid displacement" causing the Mapbox component to malfunction. They initially tried to address the problem by using the NativeUserLocation component. However, this solution didn't work as expected.

Upon further investigation, the user realized that the problem lay in the integration of the NativeUserLocation component, which was not properly nested within the Mapbox component. To resolve the issue, they moved the NativeUserLocation component inside the Mapbox component.

This change allowed their application to compile and run. However, a new problem emerged. With the NativeUserLocation component now residing within the Mapbox component, the user had limited access to the parameters of the UserLocation. This limitation resulted from the fact that the NativeUserLocation component provided only two parameters.

In summary, the user restructured their code by placing the NativeUserLocation component inside the Mapbox component, enabling the app to compile and run. However, this change came at the cost of reduced access to UserLocation parameters.

### Notes / preliminary analysis

![error_RN](https://github.com/rnmapbox/maps/assets/91210099/31e2d427-353a-404e-9311-d4cf82afd755)

**Comment by mfazekas** [comment]
@RayanCardoso  sorry I don't really understand the bugerport. Does the issue happen with or without `react-native-geolocation-service`?

**Comment by mfazekas** [comment]
Sound like duplicate of: #2717
